src/Rules/Functions/PrintfHelper.php

```diff
@@ -0,0 +1,75 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Rules\Functions;
+
+use Nette\Utils\Strings;
+use PHPStan\Php\PhpVersion;
+use function array_filter;
+use function count;
+use function max;
+use function sprintf;
+use function strlen;
+use const PREG_SET_ORDER;
+
+final class PrintfHelper
+{
+
+	public function __construct(private PhpVersion $phpVersion)
+	{
+	}
+
+	public function getPrintfPlaceholdersCount(string $format): int
+	{
+		return $this->getPlaceholdersCount('(?:[bs%s]|l?[cdeEgfFGouxX])', $format);
+	}
+
+	public function getScanfPlaceholdersCount(string $format): int
+	{
+		return $this->getPlaceholdersCount('(?:[cdDeEfinosuxX%s]|\[[^\]]+\])', $format);
+	}
```

Comment:
I need this use-cases separated because I only need the `getPrintfPlaceholdersCount` in `PrintfArrayParametersRule`